catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java

```diff
@@ -189,7 +189,7 @@ private Object executeRemoveAllFromStore() throws Exception {
             console.flush();
 
             // Break out if there are no more records to delete
-            if (amountDeleted < batchSize || batchSize < 1) {
+            if (response.getResults().isEmpty() || batchSize < 1) {
```

Comment:
I looked further up the code and I see we are in a while loop with the condition `response.getResults().size() > 0`. The Response is not modified in the loop, so I don't see this first condition `response.getResults().isEmpty()` ever happening unless I'm missing some piece of the logic.

The older logic here makes a little more sense to me, since if the amount of metacards it deleted is less than the batch size, then there are no more results and there is no point in sending out another query.

And actually I don't think there's any point in having the batchSize condition in here. It should be in the condition for the while loop, since the batch size never changes and it seems like a batchSize of 0 should essentially be a no-op, right? Right now I think it will delete one page of results and that's it. I'm not sure if that's the intended behavior